docs/project/analyzers.md

```diff
@@ -1,27 +1,23 @@
 ## Adding new analyzers to the build
 
-This repo relies on [.NET Compiler Platform analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019) to help validate the correctness, performance, and maintainability of the code.  Several existing analyzer packages are wired into the build, but it is easy to augment the utilized packages in order to experiment with additional analyzers.
+This repo relies on [.NET Compiler Platform analyzers](https://learn.microsoft.com/visualstudio/code-quality/roslyn-analyzers-overview) to help validate the correctness, performance, and maintainability of the code.  Several existing analyzer packages are wired into the build, but it is easy to augment the utilized packages in order to experiment with additional analyzers.
 
 To add an analyzer package to the build:
-1. Select a package you want to employ, for example https://www.nuget.org/packages/SonarAnalyzer.CSharp/.  This analyzer package's name is `SonarAnalyzer.CSharp` and the latest version as of this edit is `7.15.0.8572`.
-2. Add a PackageReference entry to https://github.com/dotnet/runtime/blob/main/eng/Analyzers.props, e.g.
-```XML
-<PackageReference Include="SonarAnalyzer.CSharp" Version="7.15.0.8572" />
-```
-3. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  Rules can be disabled by adding entries to the https://github.com/dotnet/runtime/blob/main/src/libraries/CodeAnalysis.ruleset file, e.g.
-```XML
-  <Rules AnalyzerId="SonarAnalyzer.CSharp" RuleNamespace="SonarAnalyzer.CSharp">
-    <Rule Id="S1006" Action="None" />
-    <Rule Id="S1075" Action="None" />
-    ...
-  </Rules>
-```
+1. Select a package you want to employ, for example https://www.nuget.org/packages/SonarAnalyzer.CSharp/.  This analyzer package's name is `SonarAnalyzer.CSharp` and the latest version as of this edit is `8.50.0.58025`.
+2. Add a `PackageReference` entry to <https://github.com/dotnet/runtime/blob/main/eng/Analyzers.props>, e.g.
+    ```XML
+    <PackageReference Include="SonarAnalyzer.CSharp" Version="SonarAnalyzerCSharpVersion" PrivateAssets="all" />
+    ```
+3. Add `<SonarAnalyzerCSharpVersion>8.50.0.58025</SonarAnalyzerCSharpVersion>` property to [Versions.props](https://github.com/dotnet/runtime/blob/main/eng/Versions.props).
```

Comment:
This should be a general statement about what's being done followed by an example, just as the previous two bullets.

That said, this splitting of the version into the separate Version.props file is really only necessary if we're actually adding the analyzer in for everyone.  If you're just adding in additional analyzers for testing locally, this is unnecessary overhead.